qiskit_experiments/visualization/drawers/mpl_drawer.py

```diff
@@ -91,17 +91,55 @@ def initialize_canvas(self):
             # especially when the analysis consists of multiple curves.
             # Inset axis is experimental implementation of matplotlib 3.0 so maybe unstable API.
             # This draws inset axes with shared x and y axis.
-            inset_ax_h = 1 / n_rows
-            inset_ax_w = 1 / n_cols
-            for i in range(n_rows):
-                for j in range(n_cols):
+            if (
+                self.figure_options.get("custom_style", {}).get("style_name", None) == "residuals"
+                and n_subplots != 2
+            ):
+                # raising an error for residual plotting that isn't on individual plot per figure.
+                raise QiskitError(
+                    "Residual plots and residual plotting style is supported for "
+                    "figures with one sub-plot only."
+                )
+
+            inset_ax_h_list = self.figure_options.custom_style.get(
+                "sub_plot_heights_list", [1 / n_rows] * n_rows
+            )
+            inset_ax_w_list = self.figure_options.custom_style.get(
+                "sub_plot_widths_list", [1 / n_cols] * n_cols
+            )
+
+            # Check that the heights and widths are lists.
+            if (not isinstance(inset_ax_h_list, List)) or (not isinstance(inset_ax_w_list, List)):
+                raise QiskitError(
+                    "Sub-plots heights and widths list need to be a list of floats that sum"
+                    " up to 1"
+                )
+
+            # adding a check for correct sizes of subplots.
+            if sum(inset_ax_h_list) != 1 or sum(inset_ax_w_list) != 1:
```

Comment:
Use `np.close` when comparing floats.

Reply:
changed as suggested to:
`if not np.isclose(sum(inset_ax_h_list), 1) or not np.isclose(sum(inset_ax_w_list), 1):`